requirements/0004-volunteer--host-identity-validated.md

```diff
@@ -0,0 +1,175 @@
+
+1. No methods of identity validation are 100% foolproof. We are using the state-of-the-art solutions, but an evil actor determined enough, can bypass _in person_ validation at the bank, let alone a remote one like ours.
+
+### Technical Risks
+
+1. Authologic Terms of Use allow up to 1 minute of downtime without notice.
+
+### Impacts
+
+None.
+
+## Data Architecture
+
+`HostVerificationSession` table SHALL be introduced, with following fields:
+- `id` an UUIDv4 compatible GUID
+- `host_id` - UUIDv4 compatible GUID of a *Host* we are verifying
+- `accommodation_id` - UUIDv4 compatible GUID of an *AccommodationUnit* that was added with the combined form
+- `conversation_id` - id of conversation on vendor side
+
+## Application Architecture
+
+### Scenarios
+
+The system needs to support the following scenarios:
+
+#### As a Host
+
+##### ... I want to offer my place for Guests
+
+System MUST allow unauthenticated user to navigate to a Registration Form.
+
+Registration Form View SHALL allow for entering *Host* information, including:
+- Contact information:
+    * Full Name
+    * E-mail
+    * Phone
+    * Call after 
+    * Call before
+- Additional Information:
+    * Comments - additional information about themselves
+    * Languages spoken
+
+Registration Form View SHALL allow for entering *AccommodationUnit* information, including:
+- Accommodation data:
+    * Address - a street name with street and apartment number, where available.
+    * City
+    * Voivodeship
+    * Zip code - also known as Postal Code.
+- Additional Information:
+    * Comments - comments about the apartment itself
+- Vacancies Information
+    * Total available places
+- Detailed Information:
+  - Pets present
+  - Pets accepted
+  - Disabled People Friendly
+  - LGBT Friendly
+  - Parking Place Available
+  - Easy Ambulance Access.
+
+Registration Form View SHALL provide a user with RODO notice on how personal data will be collected, stored and used.
+
+Registration Form View SHALL provide a user with notice that registration requires identity verification.
+
+Registration Form View SHALL include "Proceed with Registration" button. When users clicks on "Proceed With Registration" button System shall redirect user to Authologic for identity verification.
+
+When redirecting user to Authologic for identity verification, system SHALL provide Authologic with link to Verification Status View.
+
+##### ... I want to check my verification status
+
+System MUST allow user to check their verification status. If verification was successful system SHALL show a thank-you message to the user. If verification is Pending or Rejected system SHALL show a generic wait message.
+
+
+##### ... I want to recover my account
+
+When user tries to register with existing phone number system SHALL show Account Recovery View.
+Account Recovery View SHALL notify user that this option is not available and to call Crisis Point's Hot Line.
+
+### User Interface
+
+No impact.
```

Comment:
There certainly will be some impact on the GUI - not sure if this section is about it. Definitely we will need to allow handling of non-authenticated routes on the frontend as currently the only allowed is "/". Also, I believe, when user is authenticated, all routes for the host registration should redirect to the Dashboard (i.e. to exclude a scenario when Volunteer would like to register a Host this way - there are proper means to do it already). This will also require introducing some abstraction over API communication, which currently assumes only a single API which is fed with Google authentication data.